include/revng/Model/VerifyHelper.h

```diff
@@ -77,10 +77,7 @@ class VerifyHelper {
   std::set<const model::TypeDefinition *> InProgress;
   bool AssertOnFail = false;
 
-  // Name builder is stored as a pointer in order to avoid an inclusion loop:
-  // NameBuilder methods return `model::Identifier`, which depends on this
-  // helper.
-  std::unique_ptr<model::NameBuilder> NameBuilder = nullptr;
+  std::optional<model::NameBuilder> NameBuilder = std::nullopt;
```

Comment:
Because we use `VerifyHelper` all over the place. And in a lot of cases, we don't care about names, think `model::Function::verify()` (the overload without arguments).

Reply:
OK, add a comment.

Reply:
Done:
```cpp
  // This field a only gets populated if this object was created with a binary,
  // as such, no name-space related checks will take place, say, inside
  // a `model::Function::verify`.
```